### DISCUSSION TITLE
Become a contributor

### DISCUSSION THREAD

**Discussion by kdk321** [OCA/OpenUpgrade#4478] [General]
Hello, I'm new to Odoo but I already see the value of OpenUpgrade and how important it can be for the community.  Back in my testing days I attended a class for Selenium Web Driver called "Become A Contributor".  How can members of the community best become contributors to this project?  I'm not sure if something already exists but it would be good if there was some documentation that gives a general overview of the process for analyzing the changes between versions and how the upgrade scripts are created.
It seems this could be a huge workload for existing maintainers and it would be good if new contributors could ramp up quicker and offload some of the work.  Existing maintainers could spend more time reviewing PRs versus doing that actual work.
Again, I'm new but would like to hear what others think.  Thanks.

**Comment by legalsylvain** [comment]
hi @kdk321. Thanks a lot for your interest for this project !

I share your analysis that this project is strategic for the community. 
At the moment, there are just a few of us maintaining this project, which is quite technical, and any contribution is welcome. 
Perhaps you could start by reading the documentation and saying whether there's any information missing? 

https://oca.github.io/OpenUpgrade/

Question : could you elaborate what is your usage of Odoo, and how do you need OpenUpgrade ? 

Thank you very much !

**Reply by kdk321** [reply]
@legalsylvain I'm using the Community version to transition from a very old on premise system.  During that transition I'm creating custom modules to support our business needs.  Eventually we may move to Odoo.sh to offload some of the infrastructure management but until that happens I need to continue to maintain the community version.  Because of that, I want to stay up to date on versions which requires me to update the database.  I also want to take on other clients in the future and give them the option to use community depending on their needs.

**Reply by rvalyi** [reply]
There is also lots of valuable learning content about OpenUpgrade in YouTube if you search "openupgrade oca" for instance.

**Reply by kdk321** [reply]
@rvalyi Thank you!  I will take a look.
